src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -0,0 +1,436 @@
+        size_t clientInfoSize = 0;
+        if (mManager.LoadCounter(fabric_idx, count, clientInfoSize) != CHIP_NO_ERROR)
+        {
+            return 0;
+        };
+        IgnoreUnusedVariable(clientInfoSize);
+        total += count;
+    }
+
+    return total;
+}
+
+bool DefaultICDClientStorage::ICDClientInfoIteratorImpl::Next(ICDClientInfo & item)
+{
+    for (; mFabricListIndex < mManager.mFabricList.size(); mFabricListIndex++)
+    {
+        if (mClientInfoVector.size() == 0)
+        {
+            size_t clientInfoSize = 0;
+            if (mManager.Load(mManager.mFabricList[mFabricListIndex], mClientInfoVector, clientInfoSize) != CHIP_NO_ERROR)
+            {
+                continue;
+            }
+            IgnoreUnusedVariable(clientInfoSize);
+        }
+        if (mClientInfoIndex < mClientInfoVector.size())
+        {
+            item = mClientInfoVector[mClientInfoIndex];
+            mClientInfoIndex++;
+            return true;
+        }
+        mClientInfoIndex = 0;
+        mClientInfoVector.clear();
+    }
+
+    return false;
+}
+
+void DefaultICDClientStorage::ICDClientInfoIteratorImpl::Release()
+{
+    mManager.mICDClientInfoIterators.ReleaseObject(this);
+}
+
+CHIP_ERROR DefaultICDClientStorage::Init(PersistentStorageDelegate * clientInfoStore, Crypto::SymmetricKeystore * keyStore)
+{
+    VerifyOrReturnError(clientInfoStore != nullptr && keyStore != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mpClientInfoStore == nullptr && mpKeyStore == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mpClientInfoStore = clientInfoStore;
+    mpKeyStore        = keyStore;
+    CHIP_ERROR err    = LoadFabricList();
+    if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        err = CHIP_NO_ERROR;
+    }
+    return err;
+}
+
+DefaultICDClientStorage::ICDClientInfoIterator * DefaultICDClientStorage::IterateICDClientInfo()
+{
+    return mICDClientInfoIterators.CreateObject(*this);
+}
+
+CHIP_ERROR DefaultICDClientStorage::LoadCounter(FabricIndex fabricIndex, size_t & count, size_t & clientInfoSize)
+{
+    Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
+    size_t len = MaxICDCounterSize();
+    ReturnErrorCodeIf(!backingBuffer.Calloc(len), CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(CanCastTo<uint16_t>(len), CHIP_ERROR_BUFFER_TOO_SMALL);
+    uint16_t length = static_cast<uint16_t>(len);
+
+    CHIP_ERROR err = mpClientInfoStore->SyncGetKeyValue(
+        DefaultStorageKeyAllocator::FabricICDClientInfoCounter(fabricIndex).KeyName(), backingBuffer.Get(), length);
+    if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        return CHIP_NO_ERROR;
+    }
+    ReturnErrorOnFailure(err);
+
+    TLV::ScopedBufferTLVReader reader(std::move(backingBuffer), len);
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+    TLV::TLVType structType;
+    ReturnErrorOnFailure(reader.EnterContainer(structType));
+    uint32_t _count = 0;
```

Comment:
Why a leading underscore?

Reply:
no special meaning here, just try to add one temporary variable with uint32_t so that I can use reader.Get()..otherwise, it would show the error when reader.Get(size_t)  during compilation in CI.